exec/java-exec/src/main/java/org/apache/drill/metastore/NonInterestingColumnsMetadata.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.drill.metastore;
+
+import org.apache.drill.common.expression.SchemaPath;
+import org.apache.drill.exec.record.metadata.ColumnMetadata;
+import org.apache.drill.exec.record.metadata.TupleMetadata;
+import java.util.Map;
+
+/**
+ * Represents a metadata for the non-interesting columns. Since the refresh command doesn't store the non-interesting
+ * columns stats in teh cache file, there is a need to mark column statistics of non-interesting as unknown to
+ * differentiate the non-interesting columns from non-existent columns.
+ */
+public class NonInterestingColumnsMetadata implements BaseMetadata {
```

Comment:
This class currently stores ColumnStatistics but for other things it returns NULL because it is implementing `BaseMetadata` which is a much broader interface.  If the other types of metadata are not applicable, you should clarify in the comments.

Reply:
Added a comment to clarify that NonInterestingColumnsMetadata is not applicable to some of the methods.